darts/models/forecasting/forecasting_model.py

```diff
@@ -353,8 +354,20 @@ def historical_forecasts(
         stride
             The number of time steps between two consecutive predictions.
         retrain
-            Whether to retrain the model for every prediction or not. Not all models support setting
-            `retrain` to `False`. Notably, this is supported by neural networks based models.
+            Whether and/or on which condition to retrain the model before predicting.
+            This parameter supports 3 different datatypes: ``bool``, (positive) ``int``, and
+            ``Callable`` (returning a ``bool``).
+            In the case of ``bool``: retrain the model at each step (`True`), or never retrains the model (`False`).
+            Not all models support setting `retrain` to `False`.
```

Comment:
Could you move these two sentences down in the `retrain` doc? And slightly modify it, e.g.,
```
Note: some models do require being retrained every time
and do not support anything else than `retrain=True`.
```